lib/appsignal/tracer.ex

```diff
@@ -49,37 +49,33 @@ defmodule Appsignal.Tracer do
   def create_span(namespace, nil, options) do
     pid = Keyword.get(options, :pid, self())
 
-    if running?() && !ignored?(pid) do
-      span =
-        case Keyword.get(options, :start_time) do
-          nil -> Span.create_root(namespace, pid)
-          timestamp -> Span.create_root(namespace, pid, timestamp)
-        end
-
-      register(span)
+    unless ignored?(pid) do
+      namespace
+      |> Span.create_root(pid, options[:start_time])
+      |> register()
     end
   end
 
   def create_span(_namespace, parent, options) do
     pid = Keyword.get(options, :pid, self())
 
-    if running?() && !ignored?(pid) do
-      span =
-        case Keyword.get(options, :start_time) do
-          nil -> Span.create_child(parent, pid)
-          timestamp -> Span.create_child(parent, pid, timestamp)
-        end
-
-      register(span)
+    unless ignored?(pid) do
+      parent
+      |> Span.create_child(pid, options[:start_time])
+      |> register()
     end
   end
 
   @doc """
   Finds the span in the registry table.
   """
-  @spec lookup(pid()) :: list()
+  @spec lookup(pid()) :: list() | nil
   def lookup(pid) do
-    if running?(), do: :ets.lookup(@table, pid)
+    try do
+      :ets.lookup(@table, pid)
+    rescue
+      ArgumentError -> []
```

Comment:
Going by the type annotation, is the rescue class meant to return `nil`, not `[]`?